src/app.ts

```diff
@@ -3,16 +3,30 @@ import cors from "cors";
 import fileRouter from "@routes/files.route";
 import studyplanRouter from "@routes/studyplans.route";
 import resourceRouter from "@routes/resources.route";
+import cacheEndpoint from "@middlewares/rediscache"; 
+import { getAllStudyPlans } from '@controllers/studyplans.controller';
 
 const app = express();
 app.get("/", (req: Request, res: Response) => {
 	res.send("Express + TypeScript Server");
 });
 
+app.get('/api/resources/study_plans', getAllStudyPlans);
```

Comment:
You can remove this line :) 

This thing is already done in 
`app.use("/api/resources", resourceRouter)`
`resourceRouter.get("/:page", getAllResources);`

The api simply fetches from the page name

https://github.com/stamford-syntax-club/stamfordcenter-backend/blob/main/src/routes/resources.route.ts#L6C1-L6C47

Reply:
i've tried without it, somehow it was not working, that's why i added that

Reply:
lemme try

Reply:
it "works" for me - meaning the program attempts to get from redis before retrieving from database